weather-forecast/src/redux/weatherSlice/reducer.ts

```diff
@@ -0,0 +1,45 @@
+import { getCurrentWeather } from "./../weatherSlice/thinks";
+import { WeatherDataType } from "./../types";
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface WeatherState {
+  weatherData: WeatherDataType;
+  error: string | null;
+}
+
+const initialState: WeatherState = {
+  weatherData: {},
+  error: "",
+};
+
+export const weatherSlice = createSlice({
+  name: "weather forecast",
+  initialState,
+  reducers: {
+    setWeatherData: (state, action: PayloadAction<WeatherDataType>) => {
+      return {
+        ...state,
+        weatherData: action.payload,
+      };
```

Comment:
When using redux-toolkit you can mutate the state.
So you can rewrite reducers as follows
```
setWeatherData: (state, action: PayloadAction<WeatherDataType>) => {
  state.weatherData = action.payload;
}
```